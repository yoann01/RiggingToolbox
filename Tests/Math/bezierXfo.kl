
require RiggingToolbox;

operator entry(){

    Xfo a, b;
    a.tr = Vec3(-5, 5, 0);
    b.tr = Vec3(5, -5, 0);

    // construct the curve
    BezierXfo curve;
    curve.tangentLength = 3.0;
    curve.pushKnot(a);
    curve.pushKnot(b);

    // project a transform in the middle
    Xfo center = curve.project(0.5);

    // project a regularly placed array
    Xfo xfos[] = curve.projectArray(20);

    report("xfos:" + xfos);
}

