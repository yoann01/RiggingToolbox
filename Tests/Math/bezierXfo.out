xfos:[{ori:{v:{x:+0.0,y:-2.980232e-8,z:-0.120215},w:+0.992747},tr:{x:-5.0,y:+5.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.596046e-7,z:-0.199017},w:+0.979995},tr:{x:-4.264346,y:+4.819182,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+2.980232e-8,y:+0.596046e-7,z:-0.314842},w:+0.949143},tr:{x:-3.618515,y:+4.414758,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+1.490116e-8,y:-0.596046e-7,z:-0.374433},w:+0.927253},tr:{x:-3.045604,y:+3.910672,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+2.980232e-8,y:+0.0,z:-0.408664},w:+0.912684},tr:{x:-2.521201,y:+3.355887,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+2.980232e-8,y:+0.0,z:-0.430045},w:+0.902807},tr:{x:-2.029116,y:+2.772117,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.0,z:-0.443972},w:+0.89604},tr:{x:-1.559146,y:+2.17035,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.0,z:-0.453011},w:+0.891504},tr:{x:-1.104102,y:+1.557196,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+1.490116e-8,y:+0.0,z:-0.458491},w:+0.888698},tr:{x:-0.658842,y:+0.936892,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+2.980232e-8,y:+0.0,z:-0.461062},w:+0.887367},tr:{x:-0.219025,y:+0.312712,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:-2.980232e-8,z:-0.461062},w:+0.887367},tr:{x:+0.219024,y:-0.31271,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.0,z:-0.458491},w:+0.888699},tr:{x:+0.658841,y:-0.936892,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+2.980232e-8,y:+2.980232e-8,z:-0.453011},w:+0.891504},tr:{x:+1.104102,y:-1.557196,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+2.980232e-8,z:-0.443972},w:+0.89604},tr:{x:+1.559146,y:-2.170349,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.0,z:-0.430045},w:+0.902807},tr:{x:+2.029116,y:-2.772117,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.0,z:-0.408664},w:+0.912684},tr:{x:+2.521201,y:-3.355886,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+2.980232e-8,y:+0.0,z:-0.374433},w:+0.927253},tr:{x:+3.045604,y:-3.910672,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.596046e-7,z:-0.314842},w:+0.949143},tr:{x:+3.618514,y:-4.414758,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:+0.0,z:-0.199018},w:+0.979995},tr:{x:+4.264346,y:-4.819182,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}},{ori:{v:{x:+0.0,y:-2.980232e-8,z:-0.120215},w:+0.992747},tr:{x:+5.0,y:-5.0,z:+0.0},sc:{x:+1.0,y:+1.0,z:+1.0}}]
